src/fromager/external_commands.py

```diff
@@ -20,7 +22,12 @@ def run(
     env = os.environ.copy()
     env.update(extra_environ)
 
-    logger.debug(
+    if network_isolation:
+        # prevent network access by creating a new network namespace that
+        # has no routing configured.
+        cmd = ["unshare", "--net", "--map-current-user"] + cmd
```

Comment:
Platforms other than linux may not have `unshare`. Even some linux hosts may not have it installed. We can deal with it in a follow-up, but we should decide how to handle that case.

We could look at the OS to set the default for the flag.

We could look for `unshare` in `$PATH` to set the default for the flag.

We could have a config file option to set the default.

We could just make the default `False`.

Other options?

Reply:
I have updated the PR to detect platform, `shutil.which('unshare')`, and if unshare command works. The default now depends on whether unshare works or not. Users can override the detection with `fromager --network-isolation` and `fromager --no-network-isolation`.